src/ride/track_paint.c

```diff
@@ -982,6 +982,278 @@ TRACK_PAINT_FUNCTION get_track_paint_function_topspin(int trackType, int directi
+	}
+
+	if (maze_entry & (1 << 6)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 14;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 19;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_WALL_INNER_NW_SE, 14, 18, 4, 10, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 0 | 1 << 1)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CORNER, 0, 0, 1, 1, 9, height, get_current_rotation());
+	}
+
+	if (maze_entry & (1 << 4 | 1 << 5)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CORNER, 0, 30, 1, 1, 9, height, get_current_rotation());
+	}
+
+	if (maze_entry & (1 << 8 | 1 << 9)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CORNER, 30, 30, 1, 1, 9, height, get_current_rotation());
+	}
+
+	if (maze_entry & (1 << 12 | 1 << 13)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CORNER, 30, 0, 1, 1, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 0 | 1 << 13 | 1 << 14)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_TOP_LEFT, 14, 0, 2, 1, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 5 | 1 << 6 | 1 << 8)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_BOTTOM_RIGHT, 14, 30, 2, 1, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 1 | 1 << 2 | 1 << 4)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 1;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_TOP_RIGHT, 0, 14, 1, 2, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 9 | 1 << 10 | 1 << 12)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 30;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_BOTTOM_LEFT, 30, 14, 1, 2, 9, height, get_current_rotation());
+	}
+
+
+	if (maze_entry & (1 << 2 | 1 << 6 | 1 << 10 | 1 << 14)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CENTER, 14, 14, 2, 2, 8, height, get_current_rotation());
+
+		RCT2_GLOBAL(0x141E9C4, uint16) = height + 12;
```

Comment:
@duncanspumpkin Can you check this (and the following) assignment? I don't think this is as it should be.